tests/test_oc4ids_tranforms.py

```diff
@@ -0,0 +1,1332 @@
+        releases[1]["planning"]["budget"]["amount"]["amount"]
+    )
+    assert output["budget"]["amount"]["amount"] == total
+    assert output["budget"]["amount"]["currency"] == releases[0]["planning"]["budget"]["amount"]["currency"]
+
+
+def test_budget_fail():
+    releases = [
+        {
+            "ocid": "ocds-213czf-1",
+            "id": "1",
+            "tag": "planning",
+            "date": "2001-02-03T04:05:06Z",
+            "planning": {"budget": {"amount": {"amount": "999", "currency": "USD"}}},
+        },
+        {
+            "ocid": "ocds-213czf-2",
+            "id": "1",
+            "tag": "planning",
+            "date": "2001-02-03T06:07:08Z",
+            "planning": {"budget": {"amount": {"amount": "6464", "currency": "EUR"}}},
+        },
+    ]
+
+    output = transforms._run_transforms(copy.deepcopy(releases), "1", transforms=[transforms.budget],)
+    # Different currencies could not be totalled
+    assert "budget" not in output
+
+
+def test_budget_approval():
+    releases = [
+        {
+            "ocid": "ocds-213czf-1",
+            "id": "1",
+            "tag": "planning",
+            "date": "2001-02-03T04:05:06Z",
+            "planning": {
+                "documents": [
+                    {"id": "doc1", "documentType": "projectPlan", "title": "A Document"},
+                    {"id": "doc2", "documentType": "budgetApproval", "title": "Another Document"},
+                ]
+            },
+        },
+    ]
+
+    output = transforms._run_transforms(copy.deepcopy(releases), "1", transforms=[transforms.budget_approval],)
+    assert output["documents"] == [releases[0]["planning"]["documents"][1]]
+
+
+def test_purpose_one():
+    releases = [
+        {
+            "ocid": "ocds-213czf-1",
+            "id": "1",
+            "tag": "planning",
+            "date": "2001-02-03T04:05:06Z",
+            "planning": {"rationale": "We were hungry."},
+        },
+    ]
+
+    output = transforms._run_transforms(copy.deepcopy(releases), "1", transforms=[transforms.purpose],)
+    assert output["purpose"] == releases[0]["planning"]["rationale"]
+
+
+def test_purpose_multiple():
+    releases = [
+        {
+            "ocid": "ocds-213czf-1",
+            "id": "1",
+            "tag": "planning",
+            "date": "2001-02-03T04:05:06Z",
+            "planning": {"rationale": "We were hungry."},
+        },
+        {
+            "ocid": "ocds-213czf-2",
+            "id": "2",
+            "tag": "planning",
+            "date": "2001-02-03T04:05:06Z",
+            "planning": {"rationale": "There are never enough post-its."},
+        },
+    ]
+
+    rationales = "<ocds-213czf-1> We were hungry.\n<ocds-213czf-2> There are never enough post-its.\n"
+
+    output = transforms._run_transforms(copy.deepcopy(releases), "1", transforms=[transforms.purpose],)
+    assert output["purpose"] == rationales
+
+
+def test_needs_assessment():
```

Comment:
Should this test also cover an input with multiple contracting processes, each with a 'needsAssessment' in `planning/documents`?